picard/ui/mainwindow.py

```diff
@@ -1258,18 +1259,17 @@ def show_more_albums(self):
     def view_info(self, default_tab=0):
         if not self.selected_objects:
             return
-        elif isinstance(self.selected_objects[0], Album):
-            album = self.selected_objects[0]
-            dialog = AlbumInfoDialog(album, self)
-        elif isinstance(self.selected_objects[0], Cluster):
-            cluster = self.selected_objects[0]
-            dialog = ClusterInfoDialog(cluster, self)
-        elif isinstance(self.selected_objects[0], Track):
-            track = self.selected_objects[0]
-            dialog = TrackInfoDialog(track, self)
+        selected = self.selected_objects[0]
+        if isinstance(selected, Album):
+            dialog_class = AlbumInfoDialog
+        elif isinstance(selected, Cluster):
+            dialog_class = ClusterInfoDialog
+        elif isinstance(selected, Track):
+            dialog_class = TrackInfoDialog
         else:
-            file = self.tagger.get_files_from_objects(self.selected_objects)[0]
-            dialog = FileInfoDialog(file, self)
+            selected = next(iter_files_from_objects(self.selected_objects))
```

Comment:
We had this issue with the previous code also, but it `self.selected_objects` guaranteed to yield a file here?

Reply:
Actually, no. Plus there's a potential race condition if `selected_objects` is modified by another thread, so the test at start of this method doesn't guarantee we have objects. Nor we have guarantee first item is a `File` item.
So it could be safer to use a `try/except` block, catching `StopIteration`.

Also, `FileInfoDialog` could raise an exception if we pass objects which aren't handled by it, but that's another issue.

```python
try:
    selected = next(iter_files_from_objects(self.selected_objects))
except StopIteration:
   return
dialog_class = FileInfoDialog
```

And at start of method, we could replace:

```python
if not self.selected_objects:
    return
selected = self.selected_objects[0]
```

with

```python
try:
    selected = self.selected_objects[0]
except IndexError:
    return
```

Reply:
See bb80e23